# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../dovecot/dovecot-auth-mech-winbind-space_users.patch
# Copyright (C) 2012 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

Fix winbind authentication when using usernames that include spaces

see: http://www.dovecot.org/list/dovecot/2012-July/067397.html

--- a/src/auth/mech-winbind.c	2012-08-01 11:03:38.251923128 +0200
+++ b/src/auth/mech-winbind.c	2012-08-01 11:03:56.939262515 +0200
@@ -240,7 +240,7 @@ do_auth_continue(struct auth_request *au
 	} else if (strcmp(token[0], "AF") == 0) {
 		const char *user, *p, *error;
 
-		user = gss_spnego ? token[2] : token[1];
+		user = t_strarray_join(gss_spnego ? token+2 : token+1, " ");
 		i_assert(user != NULL);
 
 		p = strchr(user, '\\');

